targetselector/running_target_selector.go

```diff
@@ -153,6 +154,16 @@ func (r runningTargetSelector) TargetIsHealthy(target string, c client.Client, i
 		if p.Status.Phase != corev1.PodRunning {
 			return errors.New("pod is not Running")
 		}
+
+		// check if pod's node is gone in the case that this was a node failure
+		if instance.Spec.NodeFailure != nil {
+			var n corev1.Node
+			if err := c.Get(context.Background(), client.ObjectKey{Name: p.Spec.NodeName}, &n); err != nil {
+				if apierrors.IsNotFound(err) {
```

Comment:
Put this in as a safeguard

Reply:
I'm afraid about the behavior of get returning an error which is not a not found error? We seem to skip it intentionally, is it not dangerous?

Reply:
I felt it would be more dangerous to return those? Returning an error here means that we can ignore the status of the injector pod and directly clean up. I wouldn't want to do that just because of a timeout or other transient error. What do you think?

Reply:
I agree with that, though we do it [here](https://github.com/DataDog/chaos-controller/pull/453/files/1c4849634e8b287e8803b86f32295db7f80c489b#diff-fbedc5033a333cbdab5a7469df68499adef8dd7b1e7e3eca5824be728c0eb268R149) so I guess we should decide to do either one or the other but not to mix behaviors (not telling that the already existing one is good)

Reply:
I changed my mind here? The sole callsite of this function already handles the different error types correctly, so I don't think extra filtering inside the function is helping us at all. I'm gonna switch this to `return err` as well to stay consistent as you'd like